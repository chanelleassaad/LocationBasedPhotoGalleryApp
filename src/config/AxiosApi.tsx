import axios from 'axios';
import Geolocation from 'react-native-geolocation-service';

export const baseURL =
  'https://6601591187c91a11641aa7a0.mockapi.io/myapp/photos';

export interface IPhoto {
  id: string;
  uri: string;
  location: string;
  latitude: number;
  longitude: number;
}

let photosData: IPhoto[] = [];

export const addPhoto = async (photoUri: string) => {
  try {
    Geolocation.getCurrentPosition(
      async position => {
        try {
          const response = await axios.get(
            `https://maps.googleapis.com/maps/api/geocode/json?latlng=${position.coords.latitude},${position.coords.longitude}&key=AIzaSyA9VuLDd9-gCTm3zpgpXe0L9fb3T39HHkw`,
          );

          if (
            response.data &&
            response.data.results &&
            response.data.results.length > 0
          ) {
            const locationName = response.data.results[0].formatted_address;
            const latitude = position.coords.latitude;
            const longitude = position.coords.longitude;

            console.log('Location name:', locationName);

            const photoData = {
              uri: photoUri,
              location: locationName,
              latitude: position.coords.latitude,
              longitude: position.coords.longitude,
            };

            const dataResponse = await axios.post(baseURL, photoData);
            console.log('Photo added:', dataResponse.data);

            // Update the local state with the new photo
            const newPhoto: IPhoto = {
              id: dataResponse.data.id,
              uri: photoUri,
              location: locationName,
              latitude: latitude,
              longitude: longitude,
            };
            photosData = [...photosData, newPhoto];
            console.log('Photos updated:', photosData);
          } else {
            console.error('No results found in the Google Maps API response');
          }
        } catch (error) {
          console.error('Failed to process Google Maps API response:', error);
        }
      },
      error => {
        console.error('Failed to get location:', error);
      },
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000},
    );
  } catch (error) {
    console.error('Failed to add photo:', error);
  }
};

export const deletePhoto = async (id: string) => {
  try {
    const response = await axios.delete(`${baseURL}/${id}`);
    photosData = photosData.filter(photo => photo.id !== id);
    console.log('Photo deleted:', response.data);
  } catch (error) {
    console.error('Failed to delete photo:', error);
  }
};

export const getAllPhotos = async () => {
  try {
    const response = await axios.get(baseURL);
    photosData = response.data;
    console.log('All photos:', photosData);
    return photosData;
  } catch (error) {
    console.error('Failed to get photos:', error);
    throw error;
  }
};

export const getPhotosData = () => {
  return photosData;
};
