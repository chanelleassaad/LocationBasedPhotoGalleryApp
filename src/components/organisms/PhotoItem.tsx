import React from 'react';
import {TouchableOpacity, Animated, Image, StyleSheet} from 'react-native';
import {IPhoto} from '../../config/AxiosApi';

interface PhotoItemProps {
  item: IPhoto;
  animation: Animated.Value;
  onLongPress: () => void;
}

const PhotoItem = ({item, animation, onLongPress}: PhotoItemProps) => {
  const handlePressIn = () => {
    Animated.spring(animation, {
      toValue: 0.8,
      useNativeDriver: true,
    }).start();
  };

  const handlePressOut = () => {
    Animated.spring(animation, {
      toValue: 1,
      useNativeDriver: true,
    }).start();
  };

  return (
    <TouchableOpacity
      testID="photo-test"
      style={styles.imageTouch}
      onLongPress={onLongPress}
      onPressIn={handlePressIn}
      onPressOut={handlePressOut}
      activeOpacity={1}>
      <Animated.View
        style={{
          transform: [{scale: animation}],
        }}>
        <Image source={{uri: item.uri}} style={styles.image} />
      </Animated.View>
    </TouchableOpacity>
  );
};

export default PhotoItem;

const styles = StyleSheet.create({
  imageTouch: {
    marginHorizontal: 5,
    padding: 4,
  },
  image: {width: 66, height: 90, borderRadius: 5},
});
