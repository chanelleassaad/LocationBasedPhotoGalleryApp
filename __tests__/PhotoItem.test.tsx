import React from 'react';
import {Animated} from 'react-native';
import {render} from '@testing-library/react-native';
import PhotoItem from '../src/components/organisms/PhotoItem';
import {IPhoto} from '../src/config/AxiosApi';

describe('PhotoItem', () => {
  it('renders PhotoItem component correctly', () => {
    const item: IPhoto = {
      id: '1',
      location: 'Some Location',
      latitude: 123.456,
      longitude: 789.012,
      uri: 'https://example.com/image.jpg',
    };
    const animation = new Animated.Value(1);
    const onPress = jest.fn();

    const tree = render(
      <PhotoItem item={item} animation={animation} onPress={onPress} />,
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
