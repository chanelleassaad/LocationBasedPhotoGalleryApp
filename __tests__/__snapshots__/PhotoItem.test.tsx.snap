// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoItem renders PhotoItem component correctly 1`] = `[Function]`;
